core-plugins/docs/Joiner-batchjoiner.md

```diff
@@ -0,0 +1,84 @@
+# Joiner
+
+
+Description
+-----------
+Joins records from one or more sources based on join keys. Supports `inner` and `outer` joins, selection and renaming of output fields.  
+
+Use Case
+--------
+The transform is used when you want to join records from one or more sources similar to joins in SQL.
```

Comment:
The transform is used when you want to combine fields from one or more input, similar to the joins in SQL.